src/components/TopLine/TopLineChart.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useRef } from "react";
+import Chart from "chart.js";
+
+import styles from "./TopLineChart.module.scss";
+
+type Props = {
+    sparklineData: any;
+    pieData: any;
+};
+
+const TopLineChart = (props: Props) => {
+    const { sparklineData, pieData } = props;
+    const divArr = [] as any;
+    for (let i = 0; i < 2; ++i) {
```

Comment:
I deeply appreciate the instinct to reduce repetitive code, but in React it's most idiomatic to avoid calling hooks inside loops — this is one of the [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html). Running hooks a variable number of times will break React, but in this particular case where the loop always runs exactly twice, you can kinda get away with it. However, for the sake of code readability you should write this as two `useEffect()` calls, and also have two different refs (e.g. `pieCanvasRef` and `sparklineCanvasRef`)